src/celeritas/em/distribution/WentzelDistribution.hh

```diff
@@ -0,0 +1,309 @@
+
+    const real_type screen_coeff = compute_screening_coefficient();
+    const real_type cos_t_max_elec = compute_max_electron_cos_t();
+
+    // Parameters for scattering of a nucleus
+    real_type form_factor_coeff = 0;
+    real_type cos_t1 = 1;
+    real_type cos_t2 = -1;
+
+    // Randomly choose if scattered off of electrons instead
+    const WentzelXsCalculator xsec(target_Z(), screen_coeff, cos_t_max_elec);
+    const real_type elec_ratio = xsec();
+    if (uniform_sample(rng) < elec_ratio)
+    {
+        // TODO: Can simplify this logic with
+        //       -1 <= cos_t_max_elec <= 1
+        cos_t1 = max(cos_t1, cos_t_max_elec);
+        cos_t2 = max(cos_t2, cos_t_max_elec);
+    }
+    else
+    {
+        // TODO: Geant has a different form momentum scale for hydrogen?
+        const real_type scale
+            = (target_Z() == 1)
+                  ? 1 / 3.097e-6
+                  : value_as<MomentumSq>(data_.form_momentum_scale);
+
+        // Set nuclear form factor
+        form_factor_coeff
+            = inc_mom_sq()
+              * fastpow((real_type)target_.atomic_mass_number().get(), 2 * 0.27)
+              / scale;
+    }
+
+    // Sample scattering angle [Fern 92] where cos(theta) = 1 + 2*mu
+    // For incident electrons / positrons, theta_min = 0 always
+    const real_type w1 = 1 - cos_t1 + 2 * screen_coeff;
+    const real_type w2 = 1 - cos_t2 + 2 * screen_coeff;
+    const real_type cos_theta = clamp(
+        1 + 2 * screen_coeff - w1 * w2 / (w1 + uniform_sample(rng) * (w2 - w1)),
+        real_type{-1},
+        real_type{1});
+
+    // Calculate rejection for fake scattering
+    // TODO: Reference?
+    MottXsCalculator mott_xsec(element_data_, inc_energy_, inc_mass_);
+    const real_type form_factor
+        = calculate_form_factor(form_factor_coeff, cos_theta);
+    const real_type g_rej = mott_xsec(cos_theta) * ipow<2>(form_factor);
+
+    if (uniform_sample(rng) > g_rej)
+    {
+        return {0, 0, 1};
+    }
+
+    // Calculate scattered vector assuming azimuthal angle is isotropic
+    const real_type sin_theta = sqrt((1 - cos_theta) * (1 + cos_theta));
+    const real_type phi = 2 * celeritas::constants::pi * uniform_sample(rng);
+    return {sin_theta * cos(phi), sin_theta * sin(phi), cos_theta};
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Calculates the form factor based on the form factor model.
+ * TODO: Reference?
+ */
+CELER_FUNCTION real_type WentzelDistribution::calculate_form_factor(
+    real_type formf, real_type cos_t) const
+{
+    switch (data_.form_factor_type)
+    {
+        case NuclearFormFactorType::Flat: {
+            // In units MeV
+            const real_type ccoef = 0.00508;
+            const real_type x = sqrt(2 * inc_mom_sq() * (1 - cos_t)) * ccoef
+                                * 2;
+            return flat_form_factor(x)
+                   * flat_form_factor(
+                       x * 0.6
```

Comment:
```suggestion
                       x * real_type(0.6)
```

Reply:
Should I go through and cast all floating point literals to `real_type`?

Reply:
Yes.